src/main/java/io/axoniq/axonserver/connector/impl/AbstractIncomingInstructionStream.java

```diff
@@ -114,6 +114,9 @@ public void onNext(IN value) {
     public void onCompleted() {
         logger.debug("Stream completed from server side");
         if (unregisterOutboundStream(instructionsForPlatform)) {
+            logger.debug("Instruction stream disconnected. Scheduling reconnect");
+            Throwable t = new StreamUnexpectedlyCompletedException("Stream unexpectedly completed by server");
+            disconnectHandler.accept(t);
```

Comment:
I'd expect a test case which would validate the `disconnectHandler` is invoked with the given `StreamUnexpectedlyCompletedException`.